### PR TITLE
chore(feed-items): introduce constants for Feed Items types

### PR DESCRIPTION
**Pull request by wpiesiak**
Introduce constants for Feed Items types and replace in all use cases

### CODE REVIEW

**Review by MateuszMamczarz — APPROVED** [review]
😮 👏 🥇

File: src/api/Versions.js [hunk 1]
```diff
@@ -249,7 +250,7 @@ class Versions extends OffsetBasedAPI {
             data: {
                 data: {
                     id: versionId,
-                    type: 'file_version',
```
Comment (jstoffan) [inline src/api/Versions.js:252]: I don't think this is desired, since versions are used outside of the Activity Feed experience. The API layer and types should drive the experiences, not the other way around.

File: src/api/Annotations.js [hunk 2]
```diff
@@ -63,7 +65,7 @@ export default class Annotations extends MarkerBasedApi {
         // We don't have entries when updating/creating an annotation
         if (!data.entries) {
             // Check if the response is a comment (result of createAnnotationReply)
-            if (data.type && data.type === 'comment') {
+            if (data.type && data.type === FEED_ITEM_TYPE_COMMENT) {
```
Comment (jstoffan) [inline src/api/Annotations.js:68]: Same comment here as versions, since annotations are used in other contexts.

**Review by jstoffan — CHANGES_REQUESTED** [review]
The data access layer should not be aware of how it's being used, at least in most cases.

File: src/api/Annotations.js [hunk 3]
```diff
@@ -118,7 +119,7 @@ export default class Annotations extends MarkerBasedApi {
             },
             file_version: {
                 id: fileVersionId,
-                type: 'file_version',
+                type: FEED_ITEM_TYPE_VERSION,
```
Comment (jstoffan) [inline src/api/Annotations.js:122]: Can we revert this change or use a non-feed related constant? The Annotations API module is used outside of the context of Feed.

File: src/api/__tests__/Comments.test.js [hunk 4]
```diff
@@ -28,7 +33,7 @@ describe('api/Comments', () => {
 
     describe('successHandler()', () => {
         const comment = {
-            type: 'comment',
+            type: FEED_ITEM_TYPE_COMMENT,
```
Comment (jstoffan) [inline src/api/__tests__/Comments.test.js:36]: Can we revert these changes to the Comments tests?

File: src/api/__tests__/Versions.test.js [hunk 5]
```diff
@@ -220,7 +220,7 @@ describe('api/Versions', () => {
                 const requestData = {
                     data: {
                         id: versionId,
-                        type: 'file_version',
+                        type: FEED_ITEM_TYPE_VERSION,
```
Comment (jstoffan) [inline src/api/__tests__/Versions.test.js:223]: Revert?

File: src/common/types/annotations.js [hunk 6]
```diff
@@ -61,7 +62,7 @@ export type Annotation = {
     status?: FeedItemStatus,
     target: Target,
     total_reply_count?: number,
-    type: 'annotation',
+    type: typeof FEED_ITEM_TYPE_ANNOTATION,
```
Comment (jstoffan) [inline src/common/types/annotations.js:65]: Can we reverse the flow of types here? This file should be the source of truth for annotations-related types rather than the other way around.

**Review by jstoffan — CHANGES_REQUESTED** [review]
We still have a lot of Feed references in areas they shouldn't be in.

File: src/elements/content-sidebar/versions/StaticVersionSidebar.js [hunk 7]
```diff
@@ -32,7 +33,7 @@ const StaticVersionsSidebar = ({ isLoading, onUpgradeClick, parentName }: Props)
         return {
             id: versionNumber,
             version_number: versionNumber,
-            type: 'file_version',
+            type: FEED_ITEM_TYPE_VERSION,
```
Comment (jstoffan) [inline src/elements/content-sidebar/versions/StaticVersionSidebar.js:36]: Can we revert, as this sidebar component isn't related to Activity Feed?

File: src/common/types/tasks.js [hunk 8]
```diff
@@ -8,6 +8,7 @@ import type {
     MarkerPaginatedCollection,
     SelectorItems,
 } from './core';
+import { FEED_ITEM_TYPE_TASK } from '../../constants';
```
Comment (jstoffan) [inline src/common/types/tasks.js:11]: Can we revert? Common types should be the source of truth, rather than using Feed in them.

File: src/elements/content-sidebar/activity-feed/annotation-thread/useRepliesAPI.js [hunk 9]
```diff
@@ -65,7 +66,7 @@ const useRepliesAPI = ({
         const replyData = {
             id: uuid,
             tagged_message: message,
-            type: 'comment',
+            type: FEED_ITEM_TYPE_COMMENT,
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/useRepliesAPI.js:69]: I think we plan to extract this component from the feed directory at some point, right? Should we keep them separate?

**Review by jstoffan — APPROVED** [review]
